### ISSUE TITLE
CreateNewBlock() : ConnectBlock failed (branch v11)

### ISSUE THREAD

**Issue by vertoe** [darkcoin/darkcoin#89]
```
(gdb) r -testnet -rescan -reindex -gen=1
Starting program: /home/vertoe/opt/darkcoin/src/qt/darkcoin-qt -testnet -rescan -reindex -gen=1
warning: Could not load shared library symbols for linux-vdso.so.1.
Do you need "set solib-search-path" or "set sysroot"?
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

[...]

[Thread 0x7fffaddf0700 (LWP 8549) exited]
[Thread 0x7fffae5f1700 (LWP 8548) exited]
terminate called after throwing an instance of 'std::runtime_error'
  what():  CreateNewBlock() : ConnectBlock failed

Program received signal SIGABRT, Aborted.
[Switching to Thread 0x7fff95ffb700 (LWP 8556)]
0x00007ffff39bea97 in raise () from /usr/lib/libc.so.6
(gdb) bt
#0  0x00007ffff39bea97 in raise () from /usr/lib/libc.so.6
#1  0x00007ffff39bfe6a in abort () from /usr/lib/libc.so.6
#2  0x00007ffff42a9fcd in __gnu_cxx::__verbose_terminate_handler() () from /usr/lib/libstdc++.so.6
#3  0x00007ffff42a7e56 in __cxxabiv1::__terminate(void (*)()) () from /usr/lib/libstdc++.so.6
#4  0x00007ffff42a7ea1 in std::terminate() () from /usr/lib/libstdc++.so.6
#5  0x00007ffff42a80b8 in __cxa_throw () from /usr/lib/libstdc++.so.6
#6  0x00000000006fba1f in CreateNewBlock (scriptPubKeyIn=...) at miner.cpp:396
#7  0x00000000006fc3a3 in CreateNewBlockWithKey (reservekey=...) at miner.cpp:480
#8  0x00000000006fc91b in BitcoinMiner (pwallet=0x7fffe872e340) at miner.cpp:545
#9  0x0000000000709e5f in boost::_bi::list1<boost::_bi::value<CWallet*> >::operator()<void (*)(CWallet*), boost::_bi::list0> (this=0x7fffe8707990, f=@0x7fffe8707988: 0x6fc847 <BitcoinMiner(CWallet*)>, a=...)
    at /usr/include/boost/bind/bind.hpp:253
#10 0x0000000000709de4 in boost::_bi::bind_t<void, void (*)(CWallet*), boost::_bi::list1<boost::_bi::value<CWallet*> > >::operator() (this=0x7fffe8707988) at /usr/include/boost/bind/bind_template.hpp:20
#11 0x0000000000709d83 in boost::detail::thread_data<boost::_bi::bind_t<void, void (*)(CWallet*), boost::_bi::list1<boost::_bi::value<CWallet*> > > >::run (this=0x7fffe87077d0)
    at /usr/include/boost/thread/detail/thread.hpp:116
#12 0x00007ffff7533f2a in ?? () from /usr/lib/libboost_thread.so.1.57.0
#13 0x00007ffff51d5314 in start_thread () from /usr/lib/libpthread.so.0
#14 0x00007ffff3a7324d in clone () from /usr/lib/libc.so.6
```

**Comment by darkcoinproject** [comment]
Fixed in 76ab102ff735c11b1298ed819bc226842ce284c9, thanks for reporting this, it was a pretty bad bug!

**Comment by vertoe** [comment]
cool. just tested, it's working in 0.11.0.5, good work.
